tools/cldr-apps/src/main/java/org/unicode/cldr/web/DBUtils.java

```diff
@@ -351,7 +351,23 @@ public static boolean hasTable(String table) {
         return getInstance().tableExists(table);
     }
 
+    private static final ConcurrentHashMap<String, Boolean> existenceCache =
+            new ConcurrentHashMap<String, Boolean>();
+
     public boolean tableExists(String table) {
+        // once exists, return true
+        // if false, try again.
+        boolean cachedValue = existenceCache.getOrDefault(table, false);
+        if (cachedValue) return cachedValue;
+        cachedValue = _tableExists(table);
```

Comment:
Is this supposed to be thread-safe, because it doesn't appear to be.

Reply:
The results only go from false to true, shouldn't ever go back to false… It's OK if two threads actually hit the database to check, that's what's happening right now… This just reduces the calls directly to the database

Reply:
In other words, two threads, trying to create the same database, already needed to be synchronized outside of this code

Reply:
Ok, makes sense.